src/test/java/org/prebid/server/bidder/richaudience/RichaudienceBidderTest.java

```diff
@@ -0,0 +1,280 @@
+        // given
+        final HttpCall<BidRequest> httpCall = givenHttpCall("Incorrect body");
+
+        // when
+        final Result<List<BidderBid>> result = bidder.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getValue()).isEmpty();
+        assertThat(result.getErrors())
+                .hasSize(1)
+                .allSatisfy(error -> {
+                    assertThat(error.getType()).isEqualTo(BidderError.Type.bad_server_response);
+                    assertThat(error.getMessage()).startsWith("Failed to decode: Unrecognized token");
+                });
+    }
+
+    @Test
+    public void makeBidsShouldReturnEmptyListIfBidResponseIsNull() {
+        // given
+        final HttpCall<BidRequest> httpCall = givenHttpCall("null");
+
+        // when
+        final Result<List<BidderBid>> result = bidder.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeBidsShouldReturnEmptyListIfBidResponseSeatBidIsNull() {
+        // given
+        final HttpCall<BidRequest> httpCall = givenHttpCall("{}");
+
+        // when
+        final Result<List<BidderBid>> result = bidder.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    private BidRequest givenBidRequest(UnaryOperator<BidRequest.BidRequestBuilder> bidRequestCustomizer) {
+        final BidRequest.BidRequestBuilder builder = BidRequest.builder()
+                .device(Device.builder().ip("Some ip.").build());
+        return bidRequestCustomizer.apply(builder).build();
+    }
+
+    private BidRequest givenBidRequest(Imp... imps) {
+        return givenBidRequest(request -> request.imp(List.of(imps)));
+    }
+
+    private BidRequest givenBidRequest(String url, Imp... imps) {
+        return givenBidRequest(request -> request.site(Site.builder().page(url).build()).imp(List.of(imps)));
+    }
+
+    private Imp givenImp(UnaryOperator<ExtImpRichaudience.ExtImpRichaudienceBuilder> extCustomizer,
+                         UnaryOperator<Imp.ImpBuilder> impCustomizer) {
+
+        final ExtImpRichaudience extImpRichaudience = extCustomizer.apply(ExtImpRichaudience.builder()).build();
+        final ObjectNode ext = mapper.valueToTree(ExtPrebid.of(null, extImpRichaudience));
+        final Imp.ImpBuilder builder = Imp.builder()
+                .banner(Banner.builder().w(21).h(9).build())
+                .ext(ext);
+        return impCustomizer.apply(builder).build();
+    }
+
+    private Imp givenImp(UnaryOperator<Imp.ImpBuilder> impCustomizer) {
+        return givenImp(identity(), impCustomizer);
+    }
+
+    private HttpCall<BidRequest> givenHttpCall(String body) {
+        final HttpResponse response = HttpResponse.of(200, null, body);
+        return HttpCall.success(null, response, null);
```

Comment:
Nitpick. Pls use \n before return statement(except situations where return statement is only the one line in method). Also pls check similar